src/App.tsx

```diff
@@ -1,24 +1,220 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  FC,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo, UpdateTodoArgs } from './types/Todo';
+import { TodoList } from './Components/TodoList';
+import {
+  createTodo,
+  getTodos,
+  deleteTodo,
+  updateTodo,
+} from './api/api.todos';
+import { FilterOption } from './enums/FilterOption';
+import { Footer } from './Components/Footer';
+import { TodoNotification } from './Components/TodoNotification';
+import { Header } from './Components/Header';
 
-const USER_ID = 0;
+const USER_ID = 10898;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState(FilterOption.All);
+  const [error, setError] = useState<string | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingItems, setLoadingItems] = useState<number[]>([]);
+  const [isLoadingAddTodo, setIsLoadingAddTodo] = useState(false);
+
+  const isAllCompleted = todos.every(todo => todo.completed);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterOption) {
+        case FilterOption.Active:
+          return !todo.completed;
+
+        case FilterOption.Completed:
+          return todo.completed;
+
+        default:
+          return FilterOption.All;
+      }
+    });
+  }, [filterOption, todos]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todosFromServer) => {
+        setError(null);
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setError('Unable to upload todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    let errorTimer: number;
+
+    if (error) {
+      errorTimer = window.setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(errorTimer);
+    };
+  }, [error]);
+
+  const addTodo = useCallback(async (title: string) => {
+    setIsLoadingAddTodo(true);
+
+    const newTodo = {
+      id: 0,
+      completed: false,
+      title,
+      userId: USER_ID,
+    };
+
+    setTempTodo({ ...newTodo });
+
+    try {
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (err) {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      setIsLoadingAddTodo(false);
+    }
```

Comment:
here you use try/catch/finally with async/await syntax, above you use chaining with then + catch, why don't you use the same approach everywhere? I suggest rewriting evrerything with modern async/await syntax